From 62e61520d7699c68dd76f9a946ca44ff094013ca Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Tue, 30 Jan 2024 11:50:23 +0530
Subject: [PATCH] pcie: qcom: Fetch resource for parf region

The struct resource field in the pcie probe
is uninitialized before trying to ioremap
the parf register space. So fetching the
resource first and getting the size.

Change-Id: I57090a25de3fb87827f2a1e20c8792fd6854dfb9
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index 50d6fa3c0db7..49a7c52aa3d8 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -1751,6 +1751,7 @@ static int qcom_pcie_probe(struct platform_device *pdev)
 	}
 
 	/* get the parf size which is needed for pcie_parf_read() */
+	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "parf");
 	if (res)
 		pcie->parf_size = resource_size(res);
 
-- 
2.34.1

